framework/CodeInterfaces/PHISICS/MaterialParser.py

```diff
@@ -0,0 +1,157 @@
+    self.inputFiles = inputFiles
+    self.tree = ET.parse(self.inputFiles)
+    self.root = self.tree.getroot()
+    self.listedDict = self.fileReconstruction(self.pertDict)
+    self.printInput()
+
+  def fileReconstruction(self, deconstructedDict):
+    """
+      Converts the formatted dictionary -> {'DENSITY|FUEL1|U235':1.30, DENSITY|FUEL2|U238':4.69} 
+      into a dictionary of dictionaries that has the format -> {'DENSITY':{'FUEL1':{'U235':1.30}, 'FUEL2':{'U238':4.69}}}
+      @ In, deconstructedDict, dictionary 
+      @ Out, reconstructedDict, nested dictionary 
+    """
+    reconstructedDict           = {}
+    perturbedIsotopes           = []
+    perturbedMaterials          = []
+    perturbedPhysicalParameters = []
+    for i in deconstructedDict.iterkeys() :
+      splittedKeywords = i.split('|')
+      perturbedIsotopes.append(splittedKeywords[2])
+      perturbedMaterials.append(splittedKeywords[1])
+      perturbedPhysicalParameters.append(splittedKeywords[0])
+    for i in xrange (0,len(perturbedPhysicalParameters)):
+      reconstructedDict[perturbedPhysicalParameters[i]] = {} 
+      for j in xrange (0,len(perturbedMaterials)):
+        reconstructedDict[perturbedPhysicalParameters[i]][perturbedMaterials[j]] = {}
+        for k in xrange (0,len(perturbedIsotopes)):
+          reconstructedDict[perturbedPhysicalParameters[i]][perturbedMaterials[j]][perturbedIsotopes[k]] = {} 
+    for typeKey, value in deconstructedDict.iteritems():
+      keyWords = typeKey.split('|')
+      reconstructedDict[keyWords[0]][keyWords[1]][keyWords[2]] = value
+    return reconstructedDict
+
+  def removeRandomlyNamedFiles(self, modifiedFile):
+    """
+      Removes the temporary file with a random name in the working directory
+      @ In, modifiedFile, string
+      @ Out, None 
+    """
+    os.remove(modifiedFile)    
+       
+  def generateRandomName(self):
+    """
+      Generates a random file name for the modified file
+      @ In, None
+      @ Out, string 
+    """
+    return str(randint(1,100000000000000))+'.xml'
+    
+  def printInput(self):
+    """
+      Prints out the pertubed Qvalues library into a file
+      @ In, None 
+      @ Out, None
+    """
+    modifiedFile = self.generateRandomName()      
+    open(modifiedFile, 'w')
+    XMLdict = self.dictFormating_from_XML_to_perturbed()
+    genericXMLdict = self.dictFormating_from_perturbed_to_generic(XMLdict)    
+    newXMLDict = self.replaceValues(genericXMLdict)    
+    templatedNewXMLdict = self.fileReconstruction(newXMLDict)
+    
+    for matXML in self.root.getiterator('mat'): 
+      for isotopeXML in matXML.findall('isotope'):
+        isotopeXML.attrib['density'] = templatedNewXMLdict.get(isotopeXML.attrib.keys()[1].upper()).get(matXML.attrib.get('id').upper()).get(isotopeXML.attrib.get('id').upper())
+        self.tree.write(modifiedFile)
+    copyfile(modifiedFile, self.inputFiles)  
```

Comment:
you can just modify the original input file. No need to create a new "scratch" file